pavelib/quality.py

```diff
@@ -230,103 +125,45 @@ def run_quality(options):
     :param: p, diff-quality will fail if the quality percentage calculated is
         below this percentage. For example, if p is set to 80, and diff-quality finds
         quality of the branch vs the compare branch is less than 80%, then this task will fail.
-        This threshold would be applied to both pep8 and pylint.
+        This threshold would be applied to both PEP8 and Pylint.
     """
 
-    # Directory to put the diff reports in.
-    # This makes the folder if it doesn't already exist.
-    dquality_dir = (Env.REPORT_DIR / "diff_quality").makedirs_p()
-    diff_quality_percentage_failure = False
+    directory_report = get_clean_report_directory('diff_quality')
+    percentage_failure = False
 
-    # Set the string, if needed, to be used for the diff-quality --compare-branch switch.
-    compare_branch = getattr(options, 'compare_branch', None)
-    compare_branch_string = ''
+    compare_branch = getattr(options, 'compare_branch', '')
     if compare_branch:
-        compare_branch_string = '--compare-branch={0}'.format(compare_branch)
+        compare_branch = "--compare-branch={0}".format(compare_branch)
 
-    # Set the string, if needed, to be used for the diff-quality --fail-under switch.
     diff_threshold = int(getattr(options, 'percentage', -1))
     percentage_string = ''
     if diff_threshold > -1:
         percentage_string = '--fail-under={0}'.format(diff_threshold)
 
-    # Generate diff-quality html report for pep8, and print to console
-    # If pep8 reports exist, use those
-    # Otherwise, `diff-quality` will call pep8 itself
-
-    pep8_files = get_violations_reports("pep8")
-    pep8_reports = u' '.join(pep8_files)
-
-    try:
-        sh(
-            "diff-quality --violations=pep8 {pep8_reports} {percentage_string} "
-            "{compare_branch_string} --html-report {dquality_dir}/diff_quality_pep8.html".format(
-                pep8_reports=pep8_reports,
-                percentage_string=percentage_string,
-                compare_branch_string=compare_branch_string,
-                dquality_dir=dquality_dir
-            )
-        )
-    except BuildFailure, error_message:
-        if is_percentage_failure(error_message):
-            diff_quality_percentage_failure = True
-        else:
-            raise BuildFailure(error_message)
-
-    # Generate diff-quality html report for pylint, and print to console
-    # If pylint reports exist, use those
-    # Otherwise, `diff-quality` will call pylint itself
+    pep8_count = pep8()
+    percentage_failure = pep8_count > 0
```

Comment:
Honestly, I think the tooling is actually part of the problem here...

Please correct me if I'm wrong, but wasn't `diff-quality` created
because in the first place because there were simply too many violations
across the codebase and we needed a way to only enforce quality
standards on the delta, to "stop the bleeding", as they say.

Now that we live in a world with (near) zero PEP8 violations, I contend
that `diff-quality` has reached the end of its line (w/r/t PEP8) and is
the wrong tool for the job here. If the idea is that our automated test
suite should be reporting PEP8 violations _anywhere_ in the codebase, we
_absolutely_ should _not_ be shoe-horning `diff-quality` into that role;
we should just invoke `pep8`.

> in the place we've trained our developers to look

Does the current Jenkins quality test infrastructure invoke PEP8? If it
does, then we should just nuke `pep8` calls from `diff-quality` and
"re-train" the masses.

If it doesn't, _that's_ the actual change we should be making here.